core/src/main/java/com/scalar/db/transaction/consensuscommit/TwoPhaseConsensusCommitManager.java

```diff
@@ -132,6 +132,11 @@ public TwoPhaseCommitTransaction join(String txId) throws TransactionException {
   @VisibleForTesting
   TwoPhaseCommitTransaction join(String txId, Isolation isolation, SerializableStrategy strategy)
       throws TransactionException {
+    // If the transaction associated with the specified transaction ID is active, resume it
+    if (isTransactionActive(txId)) {
```

Comment:
Just a question.
It looks like it creates a new transaction whether the transaction associated with the specified transaction ID already exists or not, but the PR note says it used to throw TransactionException. Do I miss something?

Reply:
Actually, if it tries to create a new transaction associated with the same transaction, the following part throws `TransactionException` with the error message `The transaction already exists`:
https://github.com/scalar-labs/scalardb/blob/master/core/src/main/java/com/scalar/db/common/ActiveTransactionManagedTwoPhaseCommitTransactionManager.java#L70C17-L70C37

Reply:
Got it. Thanks!